### PR TITLE
fix get sensors belonging to node id

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

Some locations might have more than one node.
So fetch stats for specific node at the specific location.

This should resolve the duplicate map data.
Will be merge once its confirmed on staging by @nerdlet .

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: sensorsafrica/api/v2/views.py [hunk 1]
```diff
@@ -255,8 +256,8 @@ def list(self, request):
 
             nodes.append(
                 {
-                    "node_moved": prev_location is not None,
-                    "prev_location": prev_location,
+                    "node_moved": new_location is not None,
+                    "new_location": new_location,
```
Comment (kilemensi) [inline sensorsafrica/api/v2/views.py]: Just confirming @karimkawambwa:

If this is now `new_location`, shouldn't it be in the `location` field i.e. where the node is currently situated? And `last_active_node.location` would then be `prev_location`.

Comment (karimkawambwa) [reply sensorsafrica/api/v2/views.py]: `location` is where the sensor is or was on the map. The data returned here is data for the sensor at the `location`. 

So the `node_moved` to  the `new_location` and the previous location is the `location`. 

The original node that is referred in `new_location` is also returned but with `node_moved` as `false`.

@kilemensi hope this explanation makes sense sense.

Comment (karimkawambwa) [reply sensorsafrica/api/v2/views.py]: I could maybe rename the variable to `moved_to`.
So you would have `node_moved: boolean` and `moved_to: {} | null`.

Comment (karimkawambwa) [reply sensorsafrica/api/v2/views.py]: any input @phillipahereza

Comment (kilemensi) [reply sensorsafrica/api/v2/views.py]: Yeah, that's a bit better @karimkawambwa. So:

 i. We return node data for the requested location regardless of whether the node has moved or not. This way we're able to show on the map *all* the locations a node has been and its readings while at that location.
 ii. We indicate that the node has moved and we provide the *new* location for that node. I'm assuming the *new* location is the current node location i.e. if a node has moved more than once, *new* location will be where it finally ended.

Yeah, I think `moved_to` is a better name that `new_location`. Do we also provide data for this `moved_to` location or a separate request needs to be made for that?

Comment (karimkawambwa) [reply sensorsafrica/api/v2/views.py]: `moved_to` only contains the location details but that node is in the nodes list.

E.g. This is data in staging:

So this node was initially at *Code for Kenya* now its `moved_to` *Holy Family Basilica*.
The node at *Holy Family Basilica* is also returned in the same call with `moved_to` as `null` and `node_moved` as `false`.

See the partial json below:

```
.
.
.,
 {
       “node_moved”: true,
       “moved_to”: {
           “name”: “Holy Family Basilica”,
           “longitude”: 36.8110788,
           “latitude”: -1.2869555,
           “city”: {
               “name”: “Nairobi”,
               “slug”: “nairobi”
           }
       },
       “location”: {
           “name”: “Code for Kenya”,
           “longitude”: 36.79087,
           “latitude”: -1.2982944,
           “city”: {
               “name”: “Nairobi”,
               “slug”: “nairobi”
           }
       },
       “last_data_received_at”: “2018-01-23T08:15:43.871642Z”,
       “stats”: [
           {
               “value_type”: “humidity”,
               “sensor_id”: 26,
               “start_datetime”: “2018-01-23T08:12:41.474108Z”,
               “end_datetime”: “2018-01-23T08:12:41.474108Z”,
.
.
.
},
{
        "node_moved": false,
        "moved_to": null,
        "location": {
            "name": "Holy Family Basilica",
            "longitude": 36.8110788,
            "latitude": -1.2869555,
            "city": {
                "name": "Nairobi",
                "slug": "nairobi"
            }
        },
        "last_data_received_at": "2018-06-27T04:15:39.884057Z",
        "stats": [
            {
                "value_type": "humidity",
                "sensor_id": 26,
                "start_datetime": "2018-06-27T04:15:39.884057Z",
                "end_datetime": "2018-06-27T04:15:39.884057Z",
.
.
.
```

Comment (karimkawambwa) [reply sensorsafrica/api/v2/views.py]: cc @chegejames @DavidLemayian this thread ☝️

File: sensorsafrica/api/v2/views.py [hunk 2]
```diff
@@ -209,20 +209,25 @@ class CityView(mixins.ListModelMixin, viewsets.GenericViewSet):
 class NodesView(viewsets.ViewSet):
     def list(self, request):
         nodes = []
-        for last_active_node in LastActiveNodes.objects.iterator():
+        # Loop througth the last active nodes
```
Comment (kilemensi) [inline sensorsafrica/api/v2/views.py]: `through`

File: sensorsafrica/api/v2/views.py [hunk 3]
```diff
@@ -242,8 +247,10 @@ def list(self, request):
                     )
                 )
 
-            if last_active_node.location.id != node.location.id:
-                prev_location = {
+            # If the last_ctive node location is not same as current node location
```
Comment (kilemensi) [inline sensorsafrica/api/v2/views.py]: `last_active`

**Pull request merged**